src/Sprites/ImageHeader.cpp

```diff
@@ -5,7 +5,7 @@
 
 ImageHeader ImageHeader::Create(int32_t width, int32_t height, uint16_t bitCount)
 {
-	CheckValidBitCount(bitCount);
+	VerifyBitCount(bitCount);
```

Comment:
Can you find any documentation on what bit depths are actually valid? I don't recall any sources that explicitly disallow certain bit depths. If anything, I would find that unexpected, as bitmaps might be used to store sensor data, where sensors can have an arbitrary precision. Think of satellite sensor data from a weather satellite. They often produce a monochrome 2D array of scalar values, which don't really correspond to colors, but are otherwise much like a monochrome bitmap. Think heat maps, or vegetation coverage, or light reflectance (which might cover multiple wavelengths).

Reply:
You bring up some interesting points. This code may be very close to supporting the writing of 'non-common' indexed BMP Bit Counts already. I have some questions to hunt down before I have an opinion on how to proceed and would like to open an issue on this for later. This branch is getting too extensive in comments to be productive.